File: llm_generated_code/ChatGPT/Python/Level2/Q5.py
def find_peak(mountains):
    left, right = 0, len(mountains) - 1

    while left < right:
        mid = (left + right) // 2
        # Check if mid is a peak
        if mountains[mid] > mountains[mid - 1] and mountains[mid] > mountains[mid + 1]:
            return mid
        # If the left neighbor is greater, move the search to the left side
        elif mountains[mid - 1] > mountains[mid]:
            right = mid - 1
        # Otherwise, move to the right side
        else:
            left = mid + 1

    # Left and right converge at a peak index
    return left

# Example usage
mountains = [1, 3, 20, 4, 1, 0]
peak_index = find_peak(mountains)
print("One of the peaks is at index:", peak_index, "with height:", mountains[peak_index])

File: llm_generated_code/ChatGPT/CPP/Level2/Q5.cpp
#include <iostream>
#include <vector>
using namespace std;

int findPeak(const vector<int>& mountains) {
    int n = mountains.size();

    if (n == 0) return -1;  // No mountains in the range

    for (int i = 0; i < n; i++) {
        // Check if the current mountain is a peak
        if ((i == 0 || mountains[i] > mountains[i - 1]) &&
            (i == n - 1 || mountains[i] > mountains[i + 1])) {
            return mountains[i];
        }
    }

    return -1;  // No peak found, though with valid input, this case shouldn't occur
}

int main() {
    vector<int> mountains = {1, 3, 20, 4, 1, 0};  // Example heights

    int peak = findPeak(mountains);

    if (peak != -1) {
        cout << "One of the peaks in the mountain range has a height of: " << peak << endl;
    } else {
        cout << "No peak found in the mountain range." << endl;
    }

    return 0;
}

